functions/vshn-postgres-func/encrypted_pvc.go

```diff
@@ -0,0 +1,72 @@
+package vshnpostgres
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/sethvargo/go-password/password"
+
+	"github.com/go-logr/logr"
+	"github.com/vshn/appcat-comp-functions/runtime"
+	vshnv1 "github.com/vshn/component-appcat/apis/vshn/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// AddPvcSecret adds a secret for the encrypted PVC for the  PostgreSQL instance.
+func AddPvcSecret(ctx context.Context, log logr.Logger, iof *runtime.Runtime, comp *vshnv1.VSHNPostgreSQL) (*vshnv1.VSHNPostgreSQL, error) {
+
+	log.Info("Check if encrypted storage is enabled")
+
+	log.V(1).Info("Transforming", "obj", iof)
+
+	// luksSecretResourceName is the resource name defined in the composition
+	// This name is different from metadata.name of the same resource
+	// The value is hardcoded in the composition for each resource and due to crossplane limitation
+	// it cannot be matched to the metadata.name
+	var luksSecretResourceName = comp.Name + "-luks-key"
+
+	encryptionSpec := comp.Spec.Parameters.Encryption
+
+	if _, err := strconv.ParseBool(encryptionSpec.Enabled); err == nil {
```

Comment:
This doesn't do what you think it does. If `encryptionSpec.Enabled` is set to `False` this will still create the secret. I'd propose the following:

```suggestion
	if enabled, err := strconv.ParseBool(encryptionSpec.Enabled); err != nil || !enabled {
	  log.Info("Encryption not enabled")
	  return
	}
```


But also why is `encryptionSpec.Enabled` a string anyways? This could directly be a boolean?